demo_nodes_cpp/src/topics/allocator_tutorial.cpp

```diff
@@ -161,6 +161,17 @@ int main(int argc, char ** argv)
     node = rclcpp::Node::make_shared("allocator_tutorial", options);
   }
 
+  auto event_callback = [](rclcpp::ResourceStatusEvent event_type) -> void
+    {
+      if (rclcpp::ResourceStatusEvent::DEADLINE_MISSED == event_type) {
+        printf("deadline event occured");
```

Comment:
As we spoke about in our meeting, I think it would be best to have separate callbacks for each type of event, so that you could pass additional information about that event clearly as arguments to the callback. For example, here it would be good to know which instance/publisher or instance/subscription had a deadline missed. For the subscription/publisher, you can perhaps automatically associate which one it was b

Reply:
Yep, we are currently working on that.

Reply:
Ok, should I hold off reviewing for now? Or do you want more early feedback?

Reply:
(I was going to do a full review, and accidentally clicked "add single comment" mid typing 🤦‍♂️

Reply:
We would like more early feedback. We believe that rcl (https://github.com/ros2/rcl/pull/396) and rmw (https://github.com/ros2/rmw/pull/167) interfaces are relatively final.

rclcpp (https://github.com/ros2/rclcpp/pull/655) is probably the most in flux, but for now can you take a look at how we're planning to have PublisherBase and SubscriptionBase to implement the Waitable interface and removing a bunch of vectors of rcl handles in AllocatorMemoryStrategy?